### ISSUE TITLE
[EN][Item] No'Kaled, the Elements of Death

### ISSUE THREAD

**Issue by Psychobird7** [Atlantiss/BugTracker#10036]
**Description**: No'Kaled's proc rates are incorrect. -- damage values seem correct. 

**Current behaviour**: 
As tested on PTR with Heroic No'Kaled(s). 
Two No'Kaleds: 1298 Procs, 10036 Melee swings. --> 12.9% proc rate. 
One No'Kaled in MH, shield in OH: 348 Procs, 5022 Melee swings. --> 6.9% proc rate. 
You can see having two No'Kaleds equipped doubles the proc rate, which it should not. 

**Expected behaviour**: 
The proc chance should be 6.5% according to Wowhead and Simcraft. 
This is correct, **if using one No'Kaled**. 

The proc rate should remain the same with two No'Kaleds equipped. 
See the following code taken from Simcraft. https://pastebin.com/NNFb5dAz 
"// Only attacks from the slot that have No'Kaled can trigger it
  // unless the action ignores slot" 
Basically, attacks performed with the MH No'Kaled proc the OH No'Kaled, and vice versa. They shouldn't. 

**Steps to reproduce the problem**:

1. Go on PTR, equip No'Kaled(s) 
2. Hit a dummy 
3. Check the proc rates

**Revision**: 5769 (5763 on PTR)

**Comment by blinkmaster** [comment]
show video proof/forum or it didnt happen :)

**Comment by Cookiemoonster** [comment]
It did happen. I was the one testing it with him on the PTR and the proc rate is not what it should be. 
On this example he used a try where we had 10036 melee hits, we also tested it up to 20000 melee hits and the result was the same every time.
We can comfirm that the damage from the procs are correct tho :)

**Comment by blinkmaster** [comment]
im just telling, that simc is not an evidence.

**Comment by Cookiemoonster** [comment]
And I just said we did it on the PTR as well. He simmed it and I did the ingame check.
He also pointed out that the proc rate on the weapons are around 6.5% on each hit, that doesn't mean it should be doubled. On the tries we did we had the total damage done of 130M - 150M, which is more then enough damage done to have the proc rate stabilized.

**Comment by blinkmaster** [comment]
Show it shouldnt be doubled with proof

**Comment by Cookiemoonster** [comment]
Then I'll ask you the same, show proof that it should be doubled. because the proc rate is at 6.5% PER hit.
This means that for every hit you do there's only 6.5% chance of getting a proc. If you do a hit with your MH it has 6.5% chance of procing, same goes for your OH, but it's still 6.5% chance of procing in total. So this is legit - Well, if you're not able to find something that says it should be doubled, that is.

**Comment by Cookiemoonster** [comment]
What does it proc of?

It does not proc of special abilities done with the other equipped weapon (e.g. (Spell #5938) for Rogues).
It does not proc of Main Gauche (Combat Rogue Mastery).
It does not proc of auto-attacks of the other equipped weapon (off-hand in the case of rogues).

**It only procs of its own auto-attacks and special abilities done with this weapon.**

Information about the proc

The proc chance is estimated at 6.5%.
The proc chance is 33% for the Shadow/Fire/Frost variations.
The proc can crit at 150% damage.
The Fire and Frost variatons scale with Enhancement Shaman's mastery.
All procs scale with Combat Rogue's Bandit's Guile.
Last edited by Ashvael on 2011/11/19 (Patch 4.2.2)

http://www.wowhead.com/item=77188/nokaled-the-elements-of-death#comments

**Comment by Psychobird7** [comment]
Simcraft is not evidence? Alright sir. 
Simcraft is *only* the most reliable evidence we can get at this point, unless someone finds a video of this specific interaction being tested. And let me tell you, good luck with that.  

Now, considering the linked Simcraft code has a section that *specifically* handles this exact interaction, section which also has a comment specifying the interaction and the way it's meant to work? the odds of them being wrong are extremely low.  

Then we have this lovely comment; http://www.wowhead.com/item=77188/nokaled-the-elements-of-death#comments:id=1559525 
It's wowhead obviously, so unfortunately it is not exactly what one would consider reliable. 
It effectively says the exact same thing as this report, and confirms the behaviour observed based on Simcraft.  

Then we also have this lovely comment from a Shaman who does not know basic math; http://www.wowhead.com/item=78481/nokaled-the-elements-of-death?ilvl=390#comments:id=1600301 
75 total procs (31+25+19=75, yes), and 1001 melee swings, of which 57 missed. 
75/944=0.07944 --> 7.944% proc rate.  -- extremely small sample size however.

**Comment by Illusie** [comment]
Well, Honestly I never Retail at all. But I played at some Cata servers before Atlantiss.

As far I remember people been used No'Kaled for MainHand **and OffHand** for a reason(s)
One of the reason(s) can be that No'Kaled proc from OH otherwise they better used other Weapon to get more stats since No'Kaled gives only (Stamina)-**Agility**, but no Haste/Mastery/Crit/Exp/Hit

I also speak from a prespective of PvP and not only from PvE. People used No'Kaled in pvp in both Hands (Enh/Rogue) otherwise they would pick Cata weapon to get more Resilience+Stat.

BTW I think Fury Warrior double Gurthalak work same way.

**Comment by Pocketerino** [comment]
Simcraft aint evidence, jk, didn't expect less from you. Private servers aint exactly something whats taken as reference Illusie, especially with how shit other cata servers that are currently out are, even more considering how shit was the cata server scene before.

**Comment by Illusie** [comment]
@Pocketerino Hi Pocket. I'm sure Private servers are not something relaying on and Retail information is always the best to provide.
Yet I don't think all the wisdom belong to Atlantiss & their player base. Other private servers that ran Cata servers had their own player base and scripters. People who played Retail and brought knowledge into those pservers. People that wanted to make the server better and accurate. The same as Atlantiss want.
This what makes Atlantiss walk on path that other already walked. And when the time will come and Atlantiss will move on to next expansion or the server will die. Other people will try to look on what we write now. I believe it would be a shame if other people in the future will call us something that better not to take so serious. Cause I believe we all share the same goal.

Now again: as far I remember - People been used No'Kaled on both hands in the other Pservers I played.
the Internet does not give good sources about how No'Kaled worked in Retail and only Blizzard can tell us the truth.

I found by writting on google various word keys search like "no'kaled both hands proc" some links.

Like this: https://www.mmo-champion.com/threads/1047727-No-Kaled-the-Elements-of-Death

bis list enhancement shaman guides mentioning No'Kaled on both hands.
https://www.mmo-champion.com/threads/1024386-Enhancement-4-3-BiS
http://www.tauri-veins.com/enhancement-shaman-wow-pve-dps-gear-loot-best-in-slot

some battle.net links

https://us.battle.net/forums/en/wow/topic/3967597855 - according to this guide: "It also hits with both weapons which would proc more No'Kaled, the Elements of Death Static Shock Windfury Weapon Flametongue Weapon & Maelstrom Weapon."
https://eu.battle.net/forums/en/wow/topic/1498670884 - "In patch 4.3 you may want to use 2x No'Kaled"
https://eu.battle.net/forums/en/wow/topic/3429907104 - Frost DK-DW x2 No'Kaled for reason / Procs


Another pserver guide of enh 4.3.4 : https://www.freakz.ro/forum/RO-Shaman-Enhancement-PvE-Guide-5.4.8-t212278.html
which mention that MH+OH BiS weapon is No'Kaled

Now read my previous message again. There is too many info that people been used No'Kaled x2 and much less info about people been used only x1 No'Kaled.
What I'm saying it must be reason(s) to use it on BOTH hands. That why I'm calling Atlantiss to not fix something is MIGHT never been broken.

Thank you.

**Comment by mati865** [comment]
`I also think Fury Warrior wearing double Gurthalak on MH+OH works same way and proc from Both hands. I call you to dig deeper and figure it out before you change things, GL.`
It used to work like that but it was nerfed to keep the same number of procs no matter if you have one or two Gulthalaks. Every Fury guide tells to use Gulthalak in MH and Ataraxis in OH.

**Comment by bijtankga** [comment]
In my opinion this issue SHOULD BE REOPEN (if not i will create new).

Links:
Link1: http://worldoflogs.com/reports/rt-g55nue10vxebfy9u/details/32/?enc=bosses&boss=55294
Link2: http://www.wowhead.com/item=77188/nokaled-the-elements-of-death#comments:id=1559525
Link3: http://www.wowhead.com/item=77188/nokaled-the-elements-of-death#comments:id=1609587
Link4: https://us.battle.net/forums/en/wow/topic/3967597855
Link5: http://worldoflogs.com/reports/u8p4br5jrobtsbaf/details/18/?s=9670&e=9847
Link6: http://worldoflogs.com/reports/rt-fci9afcoxlpkki58/details/0/?s=7282&e=7515


As you can see this shaman have TWO No'Kaled, from norm and from HC (2x Flameblast, Shadowblast, Iceblast)

a)
Auto Attacks (AA): 552
Shadowblast: 8 + 11 = 19
Iceblast: 9 +11 = 20
Flameblast: 12 + 16 = 28
Sum from HC: 8 + 9 + 12 = 29
Sum from norm: 11 + 11 + 16 = 38
All proc: 67

**Stupid proc calculation: 67 / 552 ~ 12%**
BUT if this is true can you tell me why so low amount from HC version?

Link2 inform us that SKILS should also proc.

b)
Total hits (wihout DoT): 1529
**Stupid proc calculation: 67 / 1529 ~ 4,4%**

Link2 also inform us that only skils using weapon.

c)
Hits from weapons (AA, Storm Strike, Lava Lash): 750
**Stupid proc calculation: 67 / 750 ~ 9%**

Link3 inform us that windfury should also proc.

d)
Hits from weapons + WF: 894
**Stupid proc calculation: 67 / 894 ~ 7,5%**

Link2 also inform us that only proc from skills done by No'Kaled. So we should divide.

Both No'Kaled versions have the same Weapon Speed so amount of AA hits should be the same (+/- 1 for last hit).
As we know, Windfury should be on MH (so proc from MH).

e)
AA: 276
Storm Strike: 55
Storm Strike OFF-Hand: 55
Lava Lash (from description OH): 44
Windfury: 72

MH hits: 276+ 55 + 72 = 475
OH hits: 276+ 55 +44 = 375
**MH proc:  29 / 475 ~ 6%
OH proc:  38 / 375 ~ 10%**

Why normal version have so high chance?
If windfury can proc so maybe Flametongue from OH also should.

f)
Flametongue: 386
OH hits: 375 + 386 = 761
**OH proc:  38 / 761 ~ 5%**

Still looks litle bad.
But maybe procs from WoWhead are wrong or Blizz change them or he have bad day?
Link4 inform us that Lava Lash hits from BOTH weapons.

g)
MH hits: 475 + 44 = 519
**MH proc: 29 / 519 ~ 5,5%**

Still bad and it is more like 5,25% instead 6,5%.
Maybe we should check ranking fights on WoL.

Link5 best Enhantment on Ultraxion (dont know if he have 1 or 2)
The same calculations as above

a)
Auto Attacks (AA): 212
Shadowblast: 7
Iceblast: 7
Flameblast: 11
All proc: 25
**Stupid proc calculation: 25 / 212 ~ 11,8%**

b)
Total hits (wihout DoT): 619
**Stupid proc calculation: 25 / 619 ~ 4%**

c)
Hits from weapons (AA, Storm Strike, Lava Lash): 268
**Stupid proc calculation: 25 / 268 ~ 9,3%**

d)
Hits from weapons + WF: 340
**Stupid proc calculation: 25 / 340~ 7,3%**


e)
AA: 106
Storm Strike: 20
Storm Strike OFF-Hand: 20
Lava Lash (from description OH): 16
Windfury: 72

MH hits: 106+20+72 = 198
OH hits: 106+20+16 = 142
**MH proc:  25 / 198 ~ 12,6%
OH proc:  25 / 142 ~ 17,6%
Tohether: 25 / ( 198+142) ~7,3%**

f)
Flametongue: 143
**OH hits: 142+143 = 285
OH proc:  25 / 285 ~ 8,7%
Tohether: 25 / (198 + 285) ~ 5,1%**

g)
MH hits: 198+16 = 214
**MH proc: 25 / 214 ~ 11,6%**
**Together: 25 / (214+285) ~ 5%**

What? 5% ?

From link6 only a) and f+g)

a)
Auto Attacks (AA): 278
Shadowblast: 11
Iceblast: 12
Flameblast: 12
All proc: 35
**Stupid proc calculation: 35 / 278~ 16,%**
Much more than 2*6.5%

f+g)
MH hits: 282
OH hits: 385
**MH proc: 35 / 282 ~ 12,4%**
**OH proc: 35 / 385 ~ 9%**
**Together: 25 / (214+285) ~ 5,2%**

So I have 2 theories not shure.

1. Basing on links should proc separately only for hits using No'Kaled with 6,5% chance.
In other words
If No'Kaled in MH every AA from MH and some hits from skills should have 6,5% to proc
If No'Kaled in OH every AA from OH and some hits from skills should have 6,5% to proc
If No'Kaled in both hands, every AA and some hits from skills should have 6,5% to proc

3. Basing on calculations the sama as above but with 5-5,5% proc chance ( 5,25% ? )

**Comment by Psychobird7** [comment]
Your entire analysis is based on flawed assumptions, hence the weird numbers you are getting.  

I've already opened a new and edited report, but please, spare us this abomination.

**Comment by Pocketerino** [comment]
Holy shit people
